packages/nameguard-js/src/nameguard-js.test.ts

```diff
@@ -0,0 +1,70 @@
+import { describe, it, expect } from "vitest";
+import { createPublicClient, http } from "viem";
+import { mainnet } from "viem/chains";
+import { createClient } from "./nameguard-js";
+
+const PROVIDER_URI_MAINNET = process.env.PROVIDER_URI_MAINNET;
```

Comment:
Suggest we take a few steps now that we're introducing environment variables to the package:

1. Create a documented .example.env file. For example https://github.com/namehash/namehash-labs/blob/main/.example.env or https://github.com/namehash/nameguard/blob/main/apps/nameguard.io/.env.example
2. It looks like we already have an environment variable defined for `PROVIDER_URI_MAINNET` in our GitHub CI: https://github.com/namehash/nameguard/settings/secrets/actions So perhaps no special action to take here?
3. Please verify these tests are automatically executed and passing on our GitHub actions CI.
4. Suggest we throw some more helpful exception in the case that an expected environment variable isn't defined. For example, please see the code here: https://github.com/namehash/namehash-labs/blob/b92b115df82230187d86db9f39ea1523b7e68247/app/api/contact-form/route.tsx#L65-L72
5. Update the README.md file for this package with a section that talks about what's needed for running the unit tests or for getting started that directs people to customizing their environment variables as needed.

Reply:
```
# Below is a list of the allowed environment variables:

# These URIs are used inside unit tests and examples for looking up on-chain data.
# Note: The following environment variables are optional.
# If they are not set, the application will default to using the free public API.
# This can cause timeouts when running tests frequently.
PROVIDER_URI_MAINNET='https://eth-mainnet.g.alchemy.com/v2/YOUR_API_KEY'
PROVIDER_URI_SEPOLIA='https://eth-sepolia.g.alchemy.com/v2/YOUR_API_KEY'
```

Reply:
```
## Running Tests

1. Set the environment variables documented in `.example.env`.

2. Build the NameGuard SDK.

    ```bash
    cd /path/to/namekit/packages/nameguard-sdk
    pnpm build
    ```

3. Install NameGuard JS dependencies.

    ```bash
    cd /path/to/namekit/packages/nameguard-js
    pnpm install
    ```

4. Run the tests.

    ```bash
    pnpm test
    ```
```